### PR TITLE
Informative error message for data import

### LINKED ISSUE #323: KeyError: 'y'
Greetings. Appreciate the great package. I'm struggling to graph_from_file to run, getting a key error (see Traceback below). Running osmnx in a clean conda environment with strict channel priority as recommended by the install docs. 

The .osm file is the raw New Zealand Geofrabrik download, clipped with osmconvert to a bounding box. Any thoughts for troubleshooting?

```python
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-6-cc08ebb90c6e> in <module>
----> 1 G = ox.graph_from_file("kaikoura_clip.osm")

~/anaconda3/envs/ox/lib/python3.7/site-packages/osmnx/core.py in graph_from_file(filename, bidirectional, simplify, retain_all, name)
   1909     # create graph using this response JSON
   1910     G = create_graph(response_jsons, bidirectional=bidirectional,
-> 1911                      retain_all=retain_all, name=name)
   1912 
   1913     # simplify the graph topology as the last step.

~/anaconda3/envs/ox/lib/python3.7/site-packages/osmnx/core.py in create_graph(response_jsons, name, retain_all, bidirectional)
   1366     # use as weight
   1367     if len(G.edges) > 0:
-> 1368         G = add_edge_lengths(G)
   1369 
   1370     return G

~/anaconda3/envs/ox/lib/python3.7/site-packages/osmnx/core.py in add_edge_lengths(G)
   1199     # first load all the edges' origin and destination coordinates as a
   1200     # dataframe indexed by u, v, key
-> 1201     coords = np.array([[u, v, k, G.nodes[u]['y'], G.nodes[u]['x'], G.nodes[v]['y'], G.nodes[v]['x']] for u, v, k in G.edges(keys=True)])
   1202     df_coords = pd.DataFrame(coords, columns=['u', 'v', 'k', 'u_y', 'u_x', 'v_y', 'v_x'])
   1203     df_coords[['u', 'v', 'k']] = df_coords[['u', 'v', 'k']].astype(np.int64)

~/anaconda3/envs/ox/lib/python3.7/site-packages/osmnx/core.py in <listcomp>(.0)
   1199     # first load all the edges' origin and destination coordinates as a
   1200     # dataframe indexed by u, v, key
-> 1201     coords = np.array([[u, v, k, G.nodes[u]['y'], G.nodes[u]['x'], G.nodes[v]['y'], G.nodes[v]['x']] for u, v, k in G.edges(keys=True)])
   1202     df_coords = pd.DataFrame(coords, columns=['u', 'v', 'k', 'u_y', 'u_x', 'v_y', 'v_x'])
   1203     df_coords[['u', 'v', 'k']] = df_coords[['u', 'v', 'k']].astype(np.int64)

KeyError: 'y'
```

### PR DESCRIPTION
**Pull request by anisotropi4**
Patch to provide more informative error message with graph_from_file data import

fixes #323 and #436

### CODE REVIEW

**Comment by gboeing** [comment]
From @anisotropi4 in #436 

#### Importing using `graph_from_file` where edge data contains missing node raises a key error 

As reported by @bradleyswilson in #323 KeyError: 'y' when poorly clipped OSM data (where edges are present without corresponding nodes) is imported using `graph_from_file` this raises a `KeyError 'y'` exception in the `add_edge_lengths`. 

#### Fails with a `KeyError: 'y'` when `G.nodes[{u,v}]` does not exist but the edge `u, v, k` does:
```python
coords = np.array([[u, v, k, G.nodes[u]['y'], G.nodes[u]['x'], G.nodes[v]['y'], G.nodes[v]['x']] for u, v, k in G.edges(keys=True)])
```
The proposed fix places this in a`try/except KeyError`block, identifies the missing node(s) referenced by edges and executes a `raise TypeError` with a more informative error message:

#### Handles the `KeyError` and raises more informative clipping error `TypeError` detailing missing nodes
```python
    try:
        coords = np.array([[u, v, k, G.nodes[u]['y'], G.nodes[u]['x'], G.nodes[v]['y'], G.nodes[v]['x']] for u, v, k in G.edges(keys=True)])
    except KeyError:
        missing_nodes = {str(i) for u, v, _ in G.edges(keys=True) if not(G.nodes[u] or G.nodes[u]) for i in (u, v) if not G.nodes[i]}
        raise TypeError('Edge missing nodes {} likely due to a clipping issue'.format(','.join(missing_nodes)))
```

File: osmnx/core.py [hunk 1]
```diff
@@ -787,7 +787,11 @@ def add_edge_lengths(G):
 
     # first load all the edges' origin and destination coordinates as a
     # dataframe indexed by u, v, key
-    coords = np.array([[u, v, k, G.nodes[u]['y'], G.nodes[u]['x'], G.nodes[v]['y'], G.nodes[v]['x']] for u, v, k in G.edges(keys=True)])
+        try:
```
Comment (gboeing) [inline osmnx/core.py:790]: This indentation error is causing the CI tests to fail.

File: osmnx/core.py [hunk 2]
```diff
@@ -787,7 +787,11 @@ def add_edge_lengths(G):
...
+        coords = np.array([[u, v, k, G.nodes[u]['y'], G.nodes[u]['x'], G.nodes[v]['y'], G.nodes[v]['x']] for u, v, k in G.edges(keys=True)])
+    except KeyError:
+        missing_nodes = {str(i) for u, v, _ in G.edges(keys=True) if not(G.nodes[u] or G.nodes[u]) for i in (u, v) if not G.nodes[i]}
+        raise TypeError('Edge missing nodes {} likely due to a clipping issue'.format(', '.join(missing_nodes)))
```
Comment (gboeing) [inline osmnx/core.py:794]: I would change "likely" to "possibly" as it's difficult to estimate the likelihood representatively across the tool's entire user base.

File: osmnx/core.py (hunk 1 above)
Comment (anisotropi4) [reply osmnx/core.py:790]: Apologies. It was a copy'n'paste error from emacs as I haven't got github integration sorted .

File: osmnx/core.py (hunk 2 above)
Comment (anisotropi4) [reply osmnx/core.py]: Done plus a reference to "edge(s)" as I found, to my cost, more than one edge may be broken

**Pull request merged**